container/shim/src/fetchers/lassie.js

```diff
@@ -167,11 +191,36 @@ async function getRequestedBlockFromCar(streamIn, streamOut, requestedCidV1, pat
         throw new Error(`Requested CID ${requestedCidV1} doesn't equal first block CID ${rootCid}.`);
       }
 
-      sendBlockResponse(streamOut, bytes, cidV1.toString())
+      sendBlockResponse(streamOut, bytes, cidV1.toString());
     } else {
       const cacheKey = cidToCacheKey(cidV1);
       blockCache.set(cacheKey, bytes);
     }
     count++;
   }
 }
+
+async function queueMetricsReport() {
+  const date = lastMetricsReportDate;
+  const canReport = !date || new Date() - date > METRICS_REPORT_INTERVAL;
```

Comment:
instead of this function, would it be simpler to have a separate interval, that just reports metrics every x ms?

Reply:
Using something like this, it would also be race free, no matter how long the promises take:

```js
(async () => {
  while (true) {
    try {
      await submitMetrics()
    } catch (err) {
      debugErr(err)
    }
    await setTimeout(interval)
  }
})()
```

This has the downside that metrics aren't reported in a regular interval, at the upside of no race conditions and controlled concurrency. Wdyt?

Reply:
It is simpler, but I generally dislike having intervals floating in the global scope. If this module gets imported then the interval starts and could start spitting errors. 
`queueMetricsReport` only gets called when a request happens, which feels cleaner to me. 

Could go either way though. @DiegoRBaquero thoughts?

Reply:
Interval is a no no for me, I normally just make the functions create a timeout to themselves to account for that. Imho, for this which is not core (unlike nginx log ingesting and submission), I prefer the queuing when it happens, and as we flush it we can switch it to self-timeout as in the rest of the repo's "intervals"

Reply:
Yes, but one could mean multiple calls at the same time, the other is one "call" on an interval, if for whatever reason there's a race condition then it could execute multiple times. 

Right now it doesn't happen because it's all sync code until the setting of `lastMetricsReportDate`, but what if the `lastMetricsReportDate` was set after an `await` ever? BOOM

Reply:
>  but what if the lastMetricsReportDate was set after an await ever? BOOM

yep that's why it's set before the await. 

@juliangruber do we have a consensus here for the requested changes? It requires approval. I'm on a time limit today >.<

Reply:
sry need to bypass your approval.

Reply:
all good please go ahead